File: MarsRover/ClientApp/app/model/Photo.ts
import {Camera} from './Camera';
import { Rover } from './Rover';

export class Photo {
    id: number;
    sol: number;
    camera: Camera;
    imageSrc: string;
    earthDate: string;
    rover: Rover; 
}

export class DefaultPhoto {

    dateString: string;
    photos: Photo[];

    constructor(date: string, pics: Photo[]) {
        this.dateString = date;
        this.photos = pics;
    }
}

File: MarsRover/Models/Photo.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace MarsRover.Models
{
    public class Photo
    {
        public int Id { get; set; }
        public int Sol { get; set; }
        public Camera Camera { get; set; }
        [JsonProperty("img_src")]
        public string ImgSrc { get; set; }

        [JsonProperty("earth_date")]
        public DateTime EarthDate { get; set; }

        public Rover Rover { get; set; }
    }

    // Just used as an easy way to wrap the list of camera since the 
    // api returns one object that wraps a list of cameras
    public class PhotoContainer
    {
        public IEnumerable<Photo> Photos { get; set; }
    }
}
